Update scalar dl manual deployment guide for auditor service on AKS

scalar-boney opened the pull request:
Update the scalar dl manual deployment guide for auditor deployment. The auditor service is an optional service.

ymorimo commented on docs/ManualDeploymentGuideScalarDLOnAzure.md:
````diff
@@ -218,11 +243,22 @@ You can uninstall Scalar DL with the following Helm commands:
+
+You can uninstall Scalar Auditor with the following Helm commands:
+
+    ```console
````
This code block is not properly formatted. 4-space indentation is not needed here.

In addition, [the another code block above](https://github.com/scalar-labs/scalar-kubernetes/pull/150/files#diff-77985a58e1a44d3a5f553f0ca5a7a99987a85eb373b09eb41c8d81de01acf8bbR244) is also not needed to be indented for the same reason. (It's formatted well because it's indented with 3 spaces.)

scalar-boney replied:
Sorry for the inconvenience.
Above specified changes are integrated e7bc2c2

feeblefakie commented on docs/ManualDeploymentGuideScalarDLAuditorOnAzure.md:
```diff
@@ -0,0 +1,145 @@
+
+## What we create
+
+![image](images/auditor-arch.png)
```
The diagram doesn't need outmost box. (The diagram is too busy with too many boxes and lines so unneeded lines should be deleted)

feeblefakie replied:
Also, there are two Azure boxes (with purple color) but what do purple boxes mean? why they are separated?
(it looks like there are two subscriptions but no, right?)
Also, if we separate these two, we should separate the client SDK as well.

feeblefakie commented on docs/ManualDeploymentGuideScalarDLAuditorOnAzure.md:
```diff
@@ -0,0 +1,141 @@
+
+### Requirements
+
+* You must create a peering between the virtual networks created for Ledger and Auditor services.
```
They are not really requirements. They are the obvious actions because we create peerings here.

What you are saying here is like this.
```
Step x. Create A.

Requirements: A must be created.
```
If we say, we can say something like this, but it can be written as steps because we are creating peerings here.

```
* You must create 3 peerings between 3 VPCs.
```


What has to be taken care of in creating peerings?

scalar-boney replied:
Thank you for your review.
Document updated with the above changes 9ea12d6

feeblefakie commented on docs/ManualDeploymentGuideScalarDLAuditorOnAzure.md:
```diff
@@ -0,0 +1,141 @@
+
+### Steps
+
+* Create the peering on the basis of [Azure official guide](https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-manage-peering) with the above requirements.
```
If we need 3 peerings, have 3 steps here.

scalar-boney replied:
Thank you for your review comments.
Document updated with the above changes 9ea12d6

feeblefakie commented on docs/ManualDeploymentGuideScalarDLAuditorOnAzure.md:
```diff
@@ -0,0 +1,141 @@
+
+* Follow step 1 to step 3 in the [Deploy Scalar DL on Azure](./ManualDeploymentGuideScalarDLOnAzure.md) guide to create an Scalar DL Auditor environment.
+
+## Step 2. Peer the Virtual Networks
```
BTW, I think we should configure security groups (or Azure equivalent ones) between VPCs but it doesn't seem to be set.
Only required ports should be open between VPCs otherwise separating them doesn't seem to mean anything.
Any reason for that?

scalar-boney replied:
Thank you for your review.
Document updated with the security group (e7bc2c2 and 9ea12d6)

scalar-boney commented on docs/ManualDeploymentGuideScalarDLAuditorOnAzure.md:
```diff
@@ -0,0 +1,141 @@
+1. Download the following Scalar DL configuration files from the [scalar-kubernetes](https://github.com/scalar-labs/scalar-kubernetes/tree/audit-guide-aks/conf) repository.
+Note that they are going to be versioned in the future, so you might want to change the branch to use a proper version.
+    * Scalardl-audit-custom-values.yaml
+    * Schema-loading-custom-values.yaml
```
Sorry for the inconvenience.
Suggestion integrated e7bc2c2

MilanCorreggio commented:
@feeblefakie @scalar-boney 
I pushed some updates PTL.
The documentation was looking clear overall most of the problem I had where from my understanding of k8s.
- I added links because after creating AKS manually the doc was not refering to any next step.
- I added more details for the Security group for any person that is not confortables setting it (Including me).
- I added `kubectl create ns monitoring` because in the doc the monitoring is setted by default and we end up with an error.( Boney helped me on this and as he propose will be nice to add some details for monitoring or change the default monitoring value)

feeblefakie commented:
@MilanCorreggio Thank you!
It's very good to suggest missing descriptions but it's not good to push modifications to someone's PR.
Please just suggest 🙇 .

scalar-boney commented:
@MilanCorreggio,

> I added links because after creating AKS manually the doc was not refering to any next step.

I created a new PR to migrate the following changes
* bcb61c9
* 4c6f9d3
* 9822c35
* 678638d
https://github.com/scalar-labs/scalar-kubernetes/pull/157

> I added more details for the Security group for any person that is not confortables setting it (Including me).

I think these changes are a bit explanatory for this particular section. So I modified like this
```
* Add new Inbound rule to restrict unwanted access to Ledger and Auditor. (By default, all subnets and ports are open between VNet after peering)
    * Add new Inbound rules to the Ledger Network Security Group to allow the Ledger Envoy LoadBalancer (50051, 50052) access from the Auditor and Client.
        * You must set a higher priority than the Ledger inbound deny rule.
    * Add new Inbound rules to the Ledger Network Security Group to deny all access from the Auditor and Client.
        * You must set a lower priority than the Ledger inbound allow rule.   
    * Add new Inbound rules to the Auditor Network Security Group to allow the Auditor Envoy LoadBalancer (40051, 40052) access from the Ledger and Client.
        * You must set a higher priority than the auditor inbound deny rule.
    * Add new Inbound rules to the Auditor Network Security Group to deny all access from the Ledger and Client.
        * You must set a lower priority than the auditor inbound allow rule.
```

Is it ok?

> I added kubectl create ns monitoring because in the doc the monitoring is setted by default and we end up with an error.( Boney helped me on this and as he propose will be nice to add some details for monitoring or change the default monitoring value)


`kubectl create ns monitoring` command does not meet all requirements for monitoring service. After that, we need to manually deploy the Prometheus so I reverted it and created another PR to fix.

https://github.com/scalar-labs/scalar-kubernetes/pull/156

MilanCorreggio reviewed (approved):
LGTM!
